### ISSUE TITLE
idea: new syntax

### ISSUE THREAD

**Issue by sam0x17** [sam0x17/docify#4]
Yandros proposed a new syntax that looks like this:

```rust
#[docify::docify]
/// some comments some comments
/// @embed("examples/samples.rs", MyCoolStruct)
/// another example:
/// @embed("src/something.rs", MyThing)
/// some closing comments
pub struct Foo;
```

Originally mentioned here: https://github.com/paritytech/substrate/pull/14672#issuecomment-1659032003

If we were to go with this syntax, we would finally be able to limit expansion of docify embeds to only be attempted if the caller is `#[cfg(doc)]` which we currently lack the context information to be able to do, and which is currently impossible in `#[doc = ..]` position but is possible for an "outer attribute" approach like this.

**Comment by yjhmelody** [comment]
Is there any way to bootstrap? Use this crate to improve documentation for itself.

**Comment by sam0x17** [comment]
Yeah I'm planning to make use of the new features within docify as they get added

**Comment by kianenigma** [comment]
similar issue https://github.com/mersinvald/aquamarine/issues/5

**Comment by sam0x17** [comment]
btw, yes it seems to be a fundamental issue in rust atm that you can't generate an inner attribute from a macro. If we could write custom inner attributes this would be different, but we cannot.

even if you put a macro like this in another crate:

```rust
macro_rules! my_macro {
    () => {
        //! hey 
    }
}
```

and then in the main crate call foo::my_macro!(); as the first line, it won't work because the macro invocation itself already puts us in item position, and there is no way to do custom inner attributes
